proof/crefine/ARM/VSpace_C.thy

```diff
@@ -1713,6 +1713,41 @@ lemma setRegister_ccorres:
   apply (auto intro: typ_heap_simps elim: obj_at'_weakenE)
   done
 
+lemma msgRegisters_ccorres:
+  "n < unat n_msgRegisters \<Longrightarrow>
+  register_from_H (ARM_H.msgRegisters ! n) = (index kernel_all_substitute.msgRegisters n)"
+  apply (simp add: kernel_all_substitute.msgRegisters_def msgRegisters_unfold fupdate_def)
+  apply (simp add: Arrays.update_def n_msgRegisters_def nth_Cons' split: if_split)
+  done
+
+(* usually when we call setMR directly, we mean to only set a registers, which will
```

Comment:
only set a registers? not sure what this means... only set a single register?

Reply:
I actually copied this from another architecture so wasn't sure of the original intention either but while looking at it now I found another version of this lemma on `ARM` in a later file, and that one had a better comment. No idea about what led to this situation in the past but I've updated all architectures to be consistent on this.